test/test-fs.c

```diff
@@ -2804,3 +2804,43 @@ TEST_IMPL(get_osfhandle_valid_handle) {
   MAKE_VALGRIND_HAPPY();
   return 0;
 }
+
+TEST_IMPL(fs_file_pos_after_op_with_offset) {
+  int r;
+
+  /* Setup. */
+  unlink("test_file");
+  loop = uv_default_loop();
+
+  r = uv_fs_open(loop,
+                 &open_req1,
+                 "test_file",
+                 O_RDWR | O_CREAT,
+                 S_IWUSR | S_IRUSR,
+                 NULL);
+  ASSERT(r >= 0);
```

Comment:
My test is modified `fs_file_open_append` test, didn't give a second though. I'll update the tests.